sdk/python/kfp/cli/cli.py

```diff
@@ -58,25 +46,11 @@ def cli(ctx: click.Context, endpoint: str, iap_client_id: str, namespace: str,
     Feature stage:
     [Alpha](https://github.com/kubeflow/pipelines/blob/07328e5094ac2981d3059314cc848fbb71437a76/docs/release/feature-stages.md#alpha)
     """
-    if ctx.invoked_subcommand in _NO_CLIENT_COMMANDS:
+    NO_CLIENT_COMMANDS = ['diagnose_me', 'components']
```

Comment:
nit: (optional) missing `components` from this list was causing a bug in the past, we may consider adding a UT to ensure we don't miss updating this list when applicable--like a new command being added which should be in this list.

Reply:
This is a good idea. I will do this in a separate PR after merging #7569. Some of the changes in #7569 will help support this.

Reply:
Good idea. I will do this as a separate PR, after #7569 is merged. #7569 has some changes that will help support this.